ui/app/src/Views/SilenceList/SilenceView.elm

```diff
@@ -100,53 +108,73 @@ editButton silence =
     let
         matchers =
             List.map (\s -> ( s.name, s.value )) silence.matchers
-    in
-    case silence.status.state of
-        -- If the silence is expired, do not edit it, but instead create a new
-        -- one with the old matchers
-        Expired ->
-            a
-                [ class "btn btn-outline-info border-0"
-                , href (newSilenceFromAlertLabels matchers)
-                ]
-                [ text "Recreate"
-                ]
 
-        _ ->
-            let
-                editUrl =
-                    String.join "/" [ "#/silences", silence.id, "edit" ]
-            in
+        editUrl =
+            -- TODO: silence.id should always be set. Can this be done nicer?
+            String.join "/" [ "#/silences", Maybe.withDefault "" silence.id, "edit" ]
```

Comment:
If the id is undefined, we could maybe generate a link to create a new silence?

Reply:
The id would never be undefined, as Alertmanager can only return Silences with ids.